### PR TITLE
add importation of administrative data via CSV

### PR DESCRIPTION
**Pull request by Lea9250**
### Status
**READY** 

new option in administrative data allows users to import  admin data from CSV

### CODE REVIEW

File: plugins/main_sections/ms_admininfo/ms_admininfo.php [hunk 1]
```diff
@@ -55,6 +58,7 @@
 $data_on[2] = $l->g(1060);
 $data_on[3] = $l->g(1701);
 $data_on[4] = $l->g(1702);
+$data_on[5] = "Import from CSV";
```
Comment (RudyLaurent) [inline plugins/main_sections/ms_admininfo/ms_admininfo.php:61]: Set "import from CSV" in language file.

File: plugins/main_sections/ms_admininfo/ms_admininfo.php [hunk 2]
```diff
@@ -376,9 +385,235 @@
...
   ));
-}
 
+} elseif ($protectedPost['onglet'] == 5) {
+    // 2nd - CSV already sent
+    if (isset($protectedPost['valid_csv'])) {
+        // create new csv obj
+        $csvObj = new CSV();
+        $protectedPost['csv_filename'] = $csvObj->saveCSV($_FILES['csv_file'], $_FILES['csv_file']['name']);
+        // open csv
```
Comment (RudyLaurent) [inline plugins/main_sections/ms_admininfo/ms_admininfo.php:395]: Add checks between states.

File: require/CSV.class.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        $this->separator = $values['tvalue']['EXPORT_SEP'];
+        $this->file_path = $values['tvalue']['TMP_DIR'];
+    }
+
+    function saveCSV($file, $newname) {
+        // save uploaded csv to dl dir
+        $info = pathinfo($file['name']);
+        // $target will be defined later in interface 
+        $target = $this->file_path."/tmp_dir/".$newname;
+        if (move_uploaded_file($file['tmp_name'], $target)) {
```
Comment (RudyLaurent) [inline require/CSV.class.php:28]: Check if the path exist and create it if it doesn't exist

File: plugins/main_sections/ms_admininfo/ms_admininfo.php (hunk 2 above)
Comment (RudyLaurent) [reply plugins/main_sections/ms_admininfo/ms_admininfo.php:395]: Add more error messages

File: plugins/main_sections/ms_admininfo/ms_admininfo.php [hunk 4]
```diff
@@ -376,9 +385,235 @@
...
+                unset($links[$key]);
+             } else {
+                 $links[$key] = "fields_".$link;
+             }
+        }
+
+        function logCSVErrors($lvl) {
+            switch ($lvl) {
+                case '1':
+                    return $error = 9202;
```
Comment (RudyLaurent) [inline plugins/main_sections/ms_admininfo/ms_admininfo.php:534]: Set the right value.

File: require/function_config_generale.php [hunk 5]
```diff
@@ -388,7 +388,7 @@ function update_default_value($POST) {
         'IT_SET_PERIM', 'IT_SET_MAIL', 'IT_SET_MAIL_ADMIN', 'SNMP', 'DOWNLOAD_REDISTRIB', 'SNMP_INVENTORY_DIFF', 'TAB_CACHE',
         'INVENTORY_CACHE_ENABLED', 'USE_NEW_SOFT_TABLES', 'WARN_UPDATE', 'INVENTORY_ON_STARTUP', 'DEFAULT_CATEGORY', 'ADVANCE_CONFIGURATION',
         'INVENTORY_SAAS_ENABLED', 'ACTIVE_NEWS', 'VULN_CVESEARCH_ENABLE','VULN_CVESEARCH_VERBOSE', 'VULN_CVESEARCH_ALL', 'VULN_CVE_EXPIRE_TIME',
-        'IPDISCOVER_LINK_TAG_NETWORK','IPDISCOVER_PURGE_OLD','IPDISCOVER_PURGE_VALIDITY_TIME');
```
Comment (RudyLaurent) [inline require/function_config_generale.php:391]: Why remove this ?

File: require/function_config_generale.php [hunk 6]
```diff
@@ -572,6 +574,9 @@ function pageGUI($advance) {
 
       $def = VARLOG_DIR . '/logs';
       ligne('LOG_DIR', $l->g(825), 'radio', array('DEFAULT' => $l->g(823) . " (" . $def . ")", 'CUSTOM' => $l->g(822), 'VALUE' => $select_log), array('HIDDEN' => 'CUSTOM', 'HIDDEN_VALUE' => $values['tvalue']['LOG_DIR'], 'SIZE' => "30%", 'MAXLENGTH' => 254, 'END' => "/logs"));
+    
+      $def = VARLIB_DIR . '/tmp_dir';
+      ligne('TMP_DIR', $l->g(9211), 'radio', array('DEFAULT' => $l->g(823) . " (" . $def . ")", 'CUSTOM' => $l->g(822), 'VALUE' => $select_tmp), array('HIDDEN' => 'CUSTOM', 'HIDDEN_VALUE' => $values['tvalue']['TMP_DIR'], 'SIZE' => "30%", 'MAXLENGTH' => 254, 'END' => "/tmp_dir"));
```
Comment (RudyLaurent) [inline require/function_config_generale.php:579]: If we select "By default ...", after the validation, the radio button "custom" is selected.

File: var.php [hunk 7]
```diff
@@ -69,11 +69,11 @@
 /**
  * OCS' MySQL database version
  */
-define('GUI_VER', '7040');
+define('GUI_VER', '7041');
 /**
  * GUI Version
  */
-define('GUI_VER_SHOW', '2.8.1');
+define('GUI_VER_SHOW', '2.8');
```
Comment (gillesdubois) [inline var.php:76]: Revert to 2.8.1 plz

File: require/function_config_generale.php [hunk 8]
```diff
@@ -798,9 +802,7 @@ function pageipdiscover($advance) {
             'IPDISCOVER_MAX_ALIVE' => 'IPDISCOVER_MAX_ALIVE',
             'IPDISCOVER_NO_POSTPONE' => 'IPDISCOVER_NO_POSTPONE',
             'IPDISCOVER_USE_GROUPS' => 'IPDISCOVER_USE_GROUPS',
-            'IPDISCOVER_LINK_TAG_NETWORK' => 'IPDISCOVER_LINK_TAG_NETWORK',
-            'IPDISCOVER_PURGE_OLD' => 'IPDISCOVER_PURGE_OLD',
-            'IPDISCOVER_PURGE_VALIDITY_TIME' => 'IPDISCOVER_PURGE_VALIDITY_TIME',
```
Comment (gillesdubois) [inline require/function_config_generale.php:803]: Revert this change

File: require/function_config_generale.php [hunk 9]
```diff
@@ -825,8 +827,6 @@ function pageipdiscover($advance) {
         ligne('IPDISCOVER_NO_POSTPONE', $l->g(747), 'radio', array(1 => 'ON', 0 => 'OFF', 'VALUE' => $values['ivalue']['IPDISCOVER_NO_POSTPONE']));
         ligne('IPDISCOVER_USE_GROUPS', $l->g(748), 'radio', array(1 => 'ON', 0 => 'OFF', 'VALUE' => $values['ivalue']['IPDISCOVER_USE_GROUPS']));
         ligne('IPDISCOVER_LINK_TAG_NETWORK', $l->g(1457), 'radio', array(1 => 'ON', 0 => 'OFF', 'VALUE' => $values['ivalue']['IPDISCOVER_LINK_TAG_NETWORK']));
-        ligne('IPDISCOVER_PURGE_OLD', $l->g(1560), 'radio', array(1 => 'ON', 0 => 'OFF', 'VALUE' => $values['ivalue']['IPDISCOVER_PURGE_OLD']));
-        ligne('IPDISCOVER_PURGE_VALIDITY_TIME', $l->g(1561), 'input', array('VALUE' => $values['ivalue']['IPDISCOVER_PURGE_VALIDITY_TIME'], 'END' => $l->g(496), 'SIZE' => 1, 'MAXLENGTH' => 3, 'JAVASCRIPT' => $numeric), '', '', $sup1);
```
Comment (gillesdubois) [inline require/function_config_generale.php:829]: Revert this change

File: require/function_config_generale.php (hunk 5 above)
Comment (gillesdubois) [inline require/function_config_generale.php:391]: Revert this changes plz
